### PR TITLE
add PageNavigator to dsr

### PR DESCRIPTION
**Pull request by PranavKrishnan007**
Screenshot 2023-01-27 at 10 53 46 PM

### CODE REVIEW

File: src/components/PageNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  page: number,
+  setPage: (no: number) => void,
+  id?: string,
+  className?: string,
+  icons?: {
+    next?: IconInputType,
+    prev?: IconInputType,
+    start?: IconInputType,
+    end?: IconInputType,
+  }
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:24]: no need for the user to customize the icons, assume the default

File: src/components/PageNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  const length = Math.floor(((totalCount + itemsPerPage - 1) / itemsPerPage));
+
+  const getPageNo = () => {
+    let list = [];
+    if (length > 1) {
+      if (length < 6)
+        Array.from({ length }, (_item, i) => {
+          list.push(i + 1);
+        });
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:42]: one liner pls

```js
Array(length).fill(0).map((_, i) => list.push(i + 1))
```

File: src/components/PageNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  return (
+      <div
+          id={id}
+          className={clsx([
+            'dsr-flex dsr-items-center dsr-justify-center dsr-text-center dsr-pt-4 page-navigator',
+            className,
+          ])}
+      >
+          <div style={{ userSelect: 'none' }}>
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:65]: `dsr-select-none`

File: src/components/PageNavigator.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                          <Icon icon={iconInput?.start ?? 'chevronsLeft'} size={18} />
+                      </Button>
+                  )}
+                  {(showControls && page > 1) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(page - 1)}>
+                          <Icon icon={iconInput?.prev ?? 'chevronLeft'} size={18} />
+                      </Button>
+                  )}
+                  {getPageNo().map((item, index) =>
+                    (
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:78]: parenthesis don't have to be in a new line

File: src/components/PageNavigator.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                        <Button
+                            variant={page === item ? 'solid' : 'outline'}
+                            color={page === item ? 'contrast' : 'secondary'}
+                            disabled={page === item}
+                            key={`page_${item}_${index}`}
+                            className={clsx([
+                              'dsr-w-12 dsr-mx-1',
+                              page === item ? 'active' : '',
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:86]: what does `active` do?

File: src/components/PageNavigator.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                  )}
+                  {(showEdges && (page + 1 < length)) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(length)}>
+                          <Icon icon={iconInput?.end ?? 'chevronsRight'} size={18} />
+                      </Button>
+                  )}
+              </div>
+              {!hideItemsPerPage &&
+              <div className="dsr-flex dsr-items-center dsr-justify-center dsr-mt-3">
+                  <div className="dsr-mr-1" style={{ width: '38px' }}>
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:107]: indentation here, along with wrapping the whole div in parenthesis

Comment (Advaith3600) [reply src/components/PageNavigator.tsx:107]: in this case use width from tailwind

File: src/components/PageNavigator.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                      <SimpleSelect
+                          value={itemsPerPage}
+                          onChange={(n) => {
+                            setPage(1); // @ts-ignore
+                            setItemsPerPage(parseInt(n));
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:112]: woah! woah! woah! kiddooooooooo!!!

ts-ignore ain't allowed

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

File: src/components/PageNavigator.tsx (hunk 1 above)
Comment (PranavKrishnan007) [reply src/components/PageNavigator.tsx:24]: done 👍🏽

File: src/components/PageNavigator.tsx (hunk 7 above)
Comment (PranavKrishnan007) [reply src/components/PageNavigator.tsx:112]: Screenshot 2023-01-28 at 12 14 25 AM

Comment (PranavKrishnan007) [reply src/components/PageNavigator.tsx:112]: Screenshot 2023-01-28 at 12 15 45 AM

Comment (PranavKrishnan007) [reply src/components/PageNavigator.tsx:112]: this is how I changed it ... is this correct?

Comment (PranavKrishnan007) [reply src/components/PageNavigator.tsx]: okie fixed 👍🏽

File: src/components/PageNavigator.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                  )}
+                  {getPageNo().map((item, index) => (
+                      <Button
+                          variant={page === item ? 'solid' : 'outline'}
+                          color={page === item ? 'contrast' : 'secondary'}
+                          disabled={page === item}
+                          key={`page_${item}_${index}`}
+                          className={clsx([
+                            'dsr-w-12 dsr-mx-1',
+                          ])}
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:77]: unnecessary `clsx`

File: src/components/PageNavigator.tsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                          onClick={() => setPage(item)}
+                      >
+                          {item}
+                      </Button>
+                  ))}
+                  {(showControls && !(page + 1 >= length)) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(page + 1)}>
+                          <Icon icon={'chevronRight'} size={18} />
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:85]: make this a string thingy, not an expression

File: src/components/PageNavigator.tsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+                      </Button>
+                  )}
+                  {(showEdges && (page + 1 < length)) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(length)}>
+                          <Icon icon={'chevronsRight'} size={18} />
```
Comment (Advaith3600) [inline src/components/PageNavigator.tsx:90]: this too

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

**Review by Advaith3600 — APPROVED** [review]

**Pull request merged**